tests/suites/test_suite_psa_crypto_slot_management.data

```diff
@@ -1,3 +1,24 @@
+Check key ID range: MBEDTLS_PSA_KEY_ID_BUILTIN_MIN
+check_key_id_value: MBEDTLS_PSA_KEY_ID_BUILTIN_MIN
+
+Check key ID range: MBEDTLS_PSA_KEY_ID_BUILTIN_MAX
+check_key_id_value: MBEDTLS_PSA_KEY_ID_BUILTIN_MAX
+
+Check key ID range: PSA_KEY_ID_USER_MIN
+check_key_id_value: PSA_KEY_ID_USER_MIN
+
+Check key ID range: PSA_KEY_ID_USER_MAX
+check_key_id_value: PSA_KEY_ID_USER_MAX
+
+Check key ID range: PSA_KEY_ID_VOLATILE_MIN
+check_key_id_value: PSA_KEY_ID_VOLATILE_MIN
+
+Check key ID range: PSA_KEY_ID_VOLATILE_MAX
+check_key_id_value: PSA_KEY_ID_VOLATILE_MAX
+
+Check key ID range: PSA_KEY_ID_VENDOR_MAX (!volatile && !built-in && !user)
```

Comment:
If we move some ranges around, `PSA_KEY_ID_VENDOR_MAX` could become part of the volatile range. I would prefer to have the test data assert which range the value fits in. That is:

```
void check_key_id_value(int key_id_arg, int is_builtin, int is_persistent)
```

and use `is_builtin` and `is_persistent` from the test data instead of calculating `builtin_attempts` and `persistent_attempts` in the test code.